Sources/CocoaLumberjack/DDFileLogger.m

```diff
@@ -1345,6 +1348,12 @@ - (void)lt_logData:(NSData *)data {
             [self willLogMessage:_currentLogFileInfo];
         }
 
+        // use an advisory lock to coordinate write with other processes
+        int fd = [handle fileDescriptor];
+        while(flock(fd, LOCK_EX) != 0) {
```

Comment:
IMHO we should make this configurable in some way.

AFAIK the current default behavior of `DDFileLogger` will prevent this kind of issue by using a different file for different processes. 
So this only happens if a custom log file manager is used. So we could actually call out to the log file manager and either ask whether to lock the file, or even a general `willWriteToFile:` and `didWriteToFile:` callback, that in turn can be used by the log file manager to lock/unlock the file.

Reply:
Well, if only one process ever accesses the logfile, the locking would be a noop and should not do any harm.

That said: I can, of course, implement it differently. Which one of your suggestions do you actually prefer?

Reply:
@ffried I was thinking about this more:

Implementing `willWriteToFile:` and `didWriteToFile:` callbacks will leave the burden to remember to lock the file to the user writing the custom LogFileManager.
I'd prefer to either add an option to turn on/off the locking (ideally default on) or even better to just don't make this configurable at all.

Why that?
Well, that way _it just works_ out of the box for users that use the same logfile for multiple processes. And has virtually no impact on users that don't do this.

Reply:
Yea, let's take it as is